### PR TITLE
ZEPPELIN-1861 Support custom interpreter.sh script to run interpreter

### PR DESCRIPTION
**Pull request by jongyoul**
### What is this PR for?
Supporting custom interpreter.sh for an individual interpreter. Some interpreters doesn't need current complicated interpreter.sh and want to use custom RemoteInterpreterServer class to launch their interpreters.

### What type of PR is it?
[Feature]

### Todos
* [x] - Add new way to be able to run custom interpreter.sh

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-1861

### How should this be tested?
Users can add 
```
"runner": {
      "linux": "interpreter.sh",
      "win": "interpreter.cmd"
    }
```
in their interpreter-setting.json. This path is relative from `${INTERPRETER_DIR}/${INTERPRETER_NAME}`

### Screenshots (if appropriate)
N/A

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by jongyoul** [comment]
I have tested md for this new way but I didn't add that code because I didn't think it's good to add that example.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java [hunk 1]
```diff
@@ -1130,15 +1135,21 @@ private Interpreter connectToRemoteRepl(String interpreterSessionKey, String cla
...
       String className, Properties property, String interpreterSettingId,
-      String userName, Boolean isUserImpersonate) {
+      String userName, Boolean isUserImpersonate, InterpreterRunner interpreterRunner) {
     int connectTimeout = conf.getInt(ConfVars.ZEPPELIN_INTERPRETER_CONNECT_TIMEOUT);
     String localRepoPath = conf.getInterpreterLocalRepoPath() + "/" + interpreterSettingId;
     int maxPoolSize = conf.getInt(ConfVars.ZEPPELIN_INTERPRETER_MAX_POOL_SIZE);
+    String interpreterRunnerPath;
+    if (null != interpreterRunner) {
+      interpreterRunnerPath = Joiner.on(File.separator)
+          .join(interpreterPath, interpreterRunner.getPath());
```
Comment (Leemoonsoo) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java]: Can  we make it accept absolute path when path starts with `/` ?

**Comment by felixcheung** [comment]
interesting. should we have documentation for this?

**Comment by jongyoul** [comment]
I made a PR for editing those values in GUI. @felixcheung I think it would be better to make a new PR to describe interpreter-setting.json fully.

**Comment by jongyoul** [comment]
Flaky test. Will merge it if there're no more discussion.

**Pull request closed without merging**